moda-contracts/src/interfaces/ICatalog.sol

```diff
@@ -0,0 +1,336 @@
+
+    /**
+     * @dev Sets the track status for a track
+     * @notice Only a caller with the Verifier role can call this function
+     * @param trackId - The id of the track
+     * @param status - The new status of the track
+     */
+    function setTrackStatus(string calldata trackId, TrackStatus status) external;
+
+    /**
+     * @dev Sets the track beneficiary for a track
+     * @param trackId - The id of the track
+     * @param newTrackBeneficiary - The new beneficiary of the track
+     */
+    function setTrackBeneficiary(string calldata trackId, address newTrackBeneficiary) external;
+
+    /**
+     * @dev Sets the track fingerprint hash for a track
+     * @param trackId - The id of the track
+     * @param fingerprintHash - The new fingerprint hash of the track
+     */
+    function setTrackFingerprintHash(
+        string calldata trackId,
+        string calldata fingerprintHash
+    ) external;
+
+    /**
+     * @dev Sets the track validation hash for a track
+     * @param trackId - The id of the track
+     * @param validationHash - The new validation hash of the track
+     */
+    function setTrackValidationHash(string calldata trackId, string calldata validationHash) external;
+
+    /**
+     * @dev Sets the track uri
+     * @notice Only the track owner can call this function
+     * @param trackId - The id of the track
+     * @param newTrackRegistrationHash - The new registration hash of the track
+     * i.e. the updated IPFS hash of the track metadata
+     */
+    function setTrackMetadata(
+        string calldata trackId,
+        string calldata newTrackRegistrationHash
+    ) external;
+
+    /**
+     * @dev Registers a Releases contract, a release can only be created from registered
+     * Releases contract and only tokens from registered Releases contract can be
+     * listed on the marketplace.
+     * @notice Only the factory can call this function
+     * @param releases - The address of the Releases contract
+     * @param releasesOwner - The owner of the Releases contract
+     */
+    function registerReleasesContract(address releases, address releasesOwner) external;
+
+    /**
+     * @dev Unregisters a Releases Contract. This will make the Releases contract
+     * unable to create releases.
+     * @param releases - The address of the Releases contract
+     */
+    function unregisterReleasesContract(address releases) external;
+
+    /**
+     * @dev Returns the owner of a Releases contract
+     * @param releases - The address of the Releases contract
+     */
+    function getReleasesOwner(address releases) external view returns (address owner);
+
+    /**
+     * @dev Add or revoke permission for a Releases contract to release a track
+     * @param trackId - The track id of the track
+     * @param releases - The address of the Releases contract
+     * @param hasPermission - The permission to set true/false
+     */
+    function grantReleasesAccessSingle(
```

Comment:
`setReleasesApproval()` is like `setApproval` for ERC721

Reply:
Done.